From 23919f28b4999145d575c065f033a6fa526ea1d7 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Thu, 20 Nov 2014 11:04:23 +0100
Subject: [PATCH 09/14] Use hostname command as fallback way of getting build
 host name

- getHostName() may fail for number of reasons, for example when
  networking is disabled or network is down.
- hostname command is a portable way of determining host name,
  which doesn't depend on networking to be available.
---
 gradle/buildReceipt.gradle | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/gradle/buildReceipt.gradle b/gradle/buildReceipt.gradle
index eb0209e..16882d1 100644
--- a/gradle/buildReceipt.gradle
+++ b/gradle/buildReceipt.gradle
@@ -109,7 +109,17 @@ task createBuildReceipt(dependsOn: determineCommitId) {
         try {
             hostName = InetAddress.localHost.hostName
         } catch (UnknownHostException e) {
-            hostName = "unknown"
+            def baos = new ByteArrayOutputStream()
+            def execResult = exec {
+                ignoreExitValue = true
+                commandLine = ["hostname"]
+                standardOutput = baos
+            }
+            if (execResult.exitValue == 0) {
+                hostName = new String(baos.toByteArray(), "utf8").trim()
+            } else {
+                hostName = "unknown"
+            }
         }
         def data = [
                 commitId: determineCommitId.commitId,
-- 
2.1.0

